src/fragmentarium/ui/fragment/TeiExport.js

```diff
@@ -0,0 +1,117 @@
+export function teiExport(fragment) {
+  var res = ''
+  const header = getHeader(
+    fragment.number,
+    fragment.collection,
+    fragment.museum.name,
+    fragment.description,
+    fragment.publication
+  )
+  const body = getBody(fragment)
+  const end = getEnd()
+
+  res += header
+  res += body
+  res += end
+
+  console.log(fragment)
+
+  return prettifyXml(res)
+}
+
+function getHeader(title, collection, museum, sourcedesc, publication) {
+  var res = ''
+  const start =
+    '<?xml version="1.0" encoding="UTF-8"?><TEI xmlns="http://www.tei-c.org/ns/1.0"><teiHeader><fileDesc>'
+  const title_stmt = '<titleStmt><title>' + title + '</title></titleStmt>'
+  const pub_stmt =
+    '<publicationStmt><p>' +
+    museum +
+    '</p><p>' +
+    collection +
+    ' Collection</p></publicationStmt>'
+  var source_desc = '<sourceDesc><p>' + sourcedesc + '</p>'
+  if (publication) source_desc += '<p>Publication: ' + publication + '</p>'
+  const end = '</sourceDesc></fileDesc></teiHeader><text><body>'
+
+  return res + start + title_stmt + pub_stmt + source_desc + end
+}
+
+function getEnd() {
+  return '</body></text></TEI>'
+}
+
+function getBody(fragment) {
+  var res = ''
+  for (var i = 0; i < fragment.text.allLines.length; i++) {
+    var line = fragment.text.allLines[i]
+
+    if (!line.displayValue) {
+      if (line.type === 'TextLine') {
+        res += '<p n="' + line.lineNumber.number + '">'
+        res += getContent(line)
+        res += '</p>'
+      } else {
+        //emptyLine => do nothing
+      }
+    } else {
+      res += '<p>'
+      res += getContent(line)
+      res += '</p>'
+    }
+  }
+
+  return res
+}
+
+function getContent(line) {
+  var res = ''
+  for (var j = 0; j < line.content.length; j++) {
+    var word = escapeXMLChars(line.content[j].cleanValue)
+    res += word
+    if (j !== line.content.length - 1) res += ' '
+  }
+  return res
+}
+
+function escapeXMLChars(word) {
+  return word.replace(/[<>&'"]/g, function (c) {
+    switch (c) {
+      case '<':
+        return '&lt;'
+      case '>':
+        return '&gt;'
+      case '&':
+        return '&amp;'
+      case "'":
+        return '&apos;'
+      case '"':
+        return '&quot;'
+    }
+  })
+}
+
+function prettifyXml(sourceXml) {
+  var xmlDoc = new DOMParser().parseFromString(sourceXml, 'application/xml')
+  var xsltDoc = new DOMParser().parseFromString(
+    [
+      '<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform">',
+      '  <xsl:strip-space elements="*"/>',
+      '  <xsl:template match="para[content-style][not(text())]">',
+      '    <xsl:value-of select="normalize-space(.)"/>',
+      '  </xsl:template>',
+      '  <xsl:template match="node()|@*">',
+      '    <xsl:copy><xsl:apply-templates select="node()|@*"/></xsl:copy>',
+      '  </xsl:template>',
+      '  <xsl:output indent="yes"/>',
+      '</xsl:stylesheet>',
+    ].join('\n'),
+    'application/xml'
+  )
+
+  var xsltProcessor = new XSLTProcessor()
```

Comment:
This breaks the tests and it is not recommended for production sites.

> This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future.

See: [XSLTProcessor](https://developer.mozilla.org/en-US/docs/Web/API/XSLTProcessor)